### PR TITLE
Edit AODN altimetry catalog

### PR DESCRIPTION
**Pull request by rfonsecadasilva**
- add NetCDFAODNStackSource driver
- Add time filter to AODN catalog

### CODE REVIEW

File: rompy/catalogs/aodn.yaml [hunk 1]
```diff
@@ -43,10 +43,23 @@ sources:
...
+       thredds_prefix: "http://thredds.aodn.org.au/thredds/dodsC/"
+       startdt: "{{startdt}}"
+       enddt: "{{enddt}}"
+    parameters:
+      geom: 
+        description: 'Geometry WKT'
+        type: str
+        default: 'POLYGON ((111.0000000000000000 -33.0000000000000000, 111.0000000000000000 -31.5000000000000000, 115.8000030517578125 -31.5000000000000000, 115.8000030517578125 -33.0000000000000000, 111.0000000000000000 -33.0000000000000000))'
+      startdt:
+        description: Forecast date
```
Comment (pbranson) [inline rompy/catalogs/aodn.yaml]: This description should be start date - that was my slackness earlier. Can you fix these descriptions in the catalog entries.

**Review by pbranson — COMMENTED** [review]
This all looks good to me. If you can just tidy up the parameter descriptions in the catalog this looks gtg! Thanks!

**Review by pbranson — APPROVED** [review]

**Comment by pbranson** [comment]
Actually is it possible for you to add a couple of tests too? Maybe a couple of pre-defined catalog entry requests where you know the length of the urlpath list and another where you know the length of the returned dataset.

File: rompy/catalogs/aodn.yaml (hunk 1 above)
Comment (rfonsecadasilva) [reply rompy/catalogs/aodn.yaml]: Sure, I fixed but I am not sure how to proceed. Should I close this PR, and submit a new one  (with new tests as per comment below)?

**Comment by rfonsecadasilva** [comment]
> Actually is it possible for you to add a couple of tests too? Maybe a couple of pre-defined catalog entry requests where you know the length of the urlpath list and another where you know the length of the returned dataset.

Sure, I have now these two new tests in test_intake. (For now, they are in a new commit only on my fork.)

**Comment by pbranson** [comment]
Thats good, thats how it works - any commits you add to your fork will come into this PR. That is why in general it is better to create a branch in your fork specific to the changes you want included in the PR

**Review by pbranson — CHANGES_REQUESTED** [review]
The tests are failing in your fork due to missing geopandas - can you add that to the requirements.txt

**Comment by pbranson** [comment]
You will also need to pull updates from main. I have fixed the github actions so that the tests get run on the PR

**Comment by rfonsecadasilva** [comment]
I added geopandas and merged upstream/main with your new changes.

**Comment by pbranson** [comment]
The tests are still failing - thats because there is a separate requirements file that is used to setup the testing environment. Probably there is a better way to configure this so that there isnt duplication, but you also need to add to: https://github.com/rfonsecadasilva/rompy/blob/main/scripts/ci/environment-pip.yml 

To see your tests you can look here:
https://github.com/rfonsecadasilva/rompy/actions

**Comment by pbranson** [comment]
And now the tests are running here so we should be able to monitor within this PR thread

**Comment by rfonsecadasilva** [comment]
> The tests are still failing - thats because there is a separate requirements file that is used to setup the testing environment. Probably there is a better way to configure this so that there isnt duplication, but you also need to add to: https://github.com/rfonsecadasilva/rompy/blob/main/scripts/ci/environment-pip.yml
> 
> To see your tests you can look here:
> https://github.com/rfonsecadasilva/rompy/actions

Thanks, that's good to know. I've included geopandas and intake-geopandas and not the tests successfully passed on my fork.


> And now the tests are running here so we should be able to monitor within this PR thread

I couldn't find new tests running here in the PR (maybe you first have to approve them before it starts testing?)

**Comment by pbranson** [comment]
Yeah that seemed to be because you hadnt made a commit to this repo before. I had to approve the CI run.

This all looks good - merging! 🎉

**Pull request merged**